src/Database/Auth/ApiKeyAuthenticator.ts

```diff
@@ -69,7 +70,7 @@ export class ApiKeyAuthenticator {
 
     return tryAuthenticate()
       .then((response: IResponse) => {
-        const body = response.body;
+        const body = response.body as IHash;
```

Comment:
About IHash:
1) If x is of type object, then:
a) ts compiler complains if we write: x.someProperty, but it doesn't complain when we write: x['someProperty'].
2) I think we both don't like second notation. 
3) Casting to any is also bad.
4) If we have type *object* it means it can be almost everything as we can't assume anything about its structure. 
5) What if you actually cast response body to some interface which depends on usage? You actually assuming that response should have some properties, so let's cast them. We use this technique when getting ajax responses in studio (for example each query response should have results + total count). If you can't cast to some well defined type, it means you can't assert given type at this point. As result it makes sense that TS complains about any property access on *object* type.
6) If possible don't use *Object*. Instead use *object*. See: http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html